src/components/MermaidDash.js

```diff
@@ -244,6 +240,27 @@ class MermaidDash extends Component {
     this.resize();
   }
 
+  fetchSitesChunk = async (params, pageNo = 1) => {
+    let siteResults = {};
+    const apiResults = await summary.get('/sites/', {
+      params: {
+        page: pageNo,
+        ...params
+      }
+    });
+    siteResults = apiResults;
+    return Object.keys(siteResults).length === 0 ? [] : siteResults.data.features;
+  };
+
+  fetchEntiresSites = async (params, pageNo = 1) => {
+    const results = await this.fetchSitesChunk(params, pageNo);
+    if (results.length === this.state.queryLimit) {
+      return [...results].concat(await this.fetchEntiresSites(params, pageNo + 1));
```

Comment:
I find it interesting that you're using both concat and spread syntax at the same time.